image_recognition_util/src/image_recognition_util/image_reader.py

```diff
@@ -0,0 +1,32 @@
+import cv2
+import os
+import fnmatch
+
+
+def read_annotated(dir_path, patterns=["*.jpg", "*.png", "*.jpeg"]):
+    """
+    Read annotated images from a directory. This reader assumes that the images in this directory are separated in
+    different directories with the label name as directory name. The method returns a generator of the label (string)
+    and the opencv image.
+    :param dir_path: The base directory we are going to write read
+    :param patterns: Patterns of the images the reader should match
+    """
+    for label in os.listdir(dir_path):
+        for root, dirs, files in os.walk(os.path.join(dir_path, label)):
+            for basename in files:
+                for pattern in patterns:
+                    if fnmatch.fnmatch(basename, pattern):
+                        yield label, cv2.imread(os.path.join(root, basename))
+
+if __name__ == "__main__":
```

Comment:
Remove this testing functionality, if you would like to test this, add a unit test in the test subdir of the package and include in the cmakelist

Reply:
The test is dependent on the filesystem it runs on, so no tests unfortunately.